tap_google_analytics/client.py

```diff
@@ -125,9 +149,10 @@ def _make_request(self, method, url, params=None, data=None):
         else:
             response = self.session.request(method, url, headers=headers, params=params)
 
-        error_message = _is_json(response) and response.json().get("error", {}).get("message")
-        if response.status_code == 400 and error_message:
-            raise Exception("400 Client Error: Bad Request, details: {}".format(error_message))
+        response_is_json = _is_json(response)
+        error_message = response_is_json and response.json().get("error", {}).get("message")
```

Comment:
Up to you, but if you are creating a variable that is used in the next line I think it's actually easier to read if it's just inlined like `error_message = _is_json(response) and response.json().get("error", {}).get("message")`.

Which brings up another point, do you really need to have a method _is_json? Or can you instead do a response_json(response) which essentially returns an Optional Json / False / nil or whatever? It feels like the boolean check is not as clean... But obviously that is a matter of opinion.

Reply:
That's a good point, the fallback style might be the way to go. I have little opinion on boolean vs. coercion (but I generally prefer to not coerce unless it's a True/False/None check).

Reply:
Choose the approach you think makes the code easier to understand.

Reply:
I trust your judgement either way